searx/utils.py

```diff
@@ -607,3 +607,30 @@ def eval_xpath_getindex(elements, xpath_spec, index, default=NOTSET):
         # to record xpath_spec
         raise SearxEngineXPathException(xpath_spec, 'index ' + str(index) + ' not found')
     return default
+
+
+class RE:
+    u"""regular expression that stores last match (like Perl's ``=~`` operator)"""
+
+    # pylint: disable=missing-function-docstring
+    def __init__(self, *args, **kwargs):
+        self.re = re.compile(*args, **kwargs)
+        self.last_match = None
```

Comment:
In the current architecture (multi thread), it creates side effect:
* the RE instance is shared in the yahoo engine between the different requests
* two requests can create:
     * write from thread 1
     * write from thread 2
     * read from thread 2
     * read from thread 1  <--- error

With async everywhere and no thread, it is technically okay, but still.

Reply:
Its a class instance like any other .. should I move the `RE_ago` into the `response` function?

https://github.com/searx/searx/pull/2640/files#diff-94618cb260d81eafbc56387ca5bc868415b94871879888e4c90f2c5f2049bc63R55